x/mongo/driver/topology/errors.go

```diff
@@ -59,3 +59,20 @@ func (e ServerSelectionError) Error() string {
 func (e ServerSelectionError) Unwrap() error {
 	return e.Wrapped
 }
+
+// WaitQueueTimeoutError represents a timeout when requesting a connection from the pool
+type WaitQueueTimeoutError struct {
+	Message string
```

Comment:
I like the idea, but the `Message` will always be the same, so I think we should remove this field and have the `Error` function automatically include the `timed out while checking out a connection from connection pool` message. We can add this functionality if we ever need consumers of this error to be able to customize the stringified message.

Reply:
Can do!